.github/workflows/snyk-scan.yml

```diff
@@ -0,0 +1,67 @@
+name: Snyk Security Vulnerability Scan
+
+on:
+  pull_request:
+  push:
+    tags:
+      - 'RELEASE-[0-9]+.[0-100]+.[0-9]+.[0-9]+-[0-9]+'
+    branches:
+      - 'master'
+
+jobs:
+  snyk_scan_test:
+    if: ${{ github.event_name == 'pull_request' }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@master
+        with:
+          fetch-depth: 0 # To fetch all commits history on branch (Refer: https://github.com/tj-actions/changed-files#usage)
+
+      - name: Check changed Deps files
+        uses: tj-actions/changed-files@v10
+        id: changed-files
+        with:
+          files: |
+            build.gradle$
```

Comment:
Does this checks only the root build.gradle? or does this definition go through the repo recursively and mathes all occurrences of build.gradle

Reply:
No it will catch all the build.gradle files, it acts as a regex here. I tested this